src/sentry/spans/consumers/process/factory.py

```diff
@@ -158,12 +155,34 @@ def _format_event_id(payload: Mapping[str, Any]) -> str:
     return ""
 
 
+def _deserialize_payload(payload: bytes) -> Mapping[str, Any]:
+    try:
+        return msgpack.unpackb(payload)
+    except msgpack.FormatError:
+        return json.loads(payload, use_rapid_json=True)
+
+
 def _process_message(message: Message[KafkaPayload]) -> KafkaPayload:
-    payload = msgpack.unpackb(message.payload.value)
+    payload = _deserialize_payload(message.payload.value)
     relay_span = payload["span"]
     relay_span["project_id"] = payload["project_id"]
     relay_span["event_id"] = _format_event_id(payload)
-    snuba_span = _build_snuba_span(relay_span)
+
+    organization_id = payload.get("organization_id")
+    retention_days = payload.get("retention_days")
+
+    if not (organization_id and retention_days):
```

Comment:
Nit: may be a personal preference but `if not organization_id or not retention_days` reads clearer than this shortened version.